gui/src/app/SPAnalysis/SPAnalysisQueryLoading.ts

```diff
@@ -96,8 +97,12 @@ export const fetchRemoteAnalysis = async (query: QueryParams) => {
     if (sampling_opts) {
         try {
             const parsed = JSON.parse(sampling_opts)
-            // TODO validate the parsed object
-            data.samplingOpts = parsed
+            if (isSamplingOpts(parsed)) {
+                data.samplingOpts = parsed
+            }
+            else {
+                console.error('Invalid sampling_opts in fetchRemoteAnalysis', parsed)
+            }
```

Comment:
If we're coming this far, it'd probably be good to include a validity check beyond just a type-guard--e.g. natural numbers where appropriate. (I'm okay with the discussion below that suggests it shouldn't be in the type guard proper, but I think if we're going to do the deserialization validity check it's worth it to go the whole way in another fn)

Reply:
I'm imagining most likely a formal parsing function, not in this file but closer to the type definition, that would replace the `JSON.parse()` line above and return either a valid `SamplingOpts` or `undefined`?